src/App.js

```diff
@@ -24,13 +24,29 @@ function App() {
         else if (currentWidth > TABLET_WIDTH) dispatch(setWidth(DESKTOP_WIDTH))
     }
 
+    const debounce = (func, wait) => {
+        let timeout
+
+        return function execute() {
+            const later = () => {
+                clearTimeout(timeout)
+                func()
+            }
+
+            clearTimeout(timeout)
+            timeout = setTimeout(later, wait)
+        }
+    }
```

Comment:
let's move to the shared folder to make it available for other components if needed